tests/Aspire.Hosting.Tests/Python/AddPythonProjectTests.cs

```diff
@@ -0,0 +1,159 @@
+
+public class AddPythonProjectTests
+{
+    [Fact]
+    public async Task AddPythonProject_SetsResourcePropertiesCorrectly()
+    {
+        using var builder = TestDistributedApplicationBuilder.Create();
+
+        var (projectDirectory, pythonExecutable, scriptName) = CreateTempPythonProject();
+        
+        builder.AddPythonProject("pythonProject", projectDirectory, scriptName);
+
+        var app = builder.Build();
+        var appModel = app.Services.GetRequiredService<DistributedApplicationModel>();
+        var executableResources = appModel.GetExecutableResources();
+
+        var pythonProjectResource = Assert.Single(executableResources);
+
+        Assert.Equal("pythonProject", pythonProjectResource.Name);
+        Assert.Equal(projectDirectory, pythonProjectResource.WorkingDirectory);
+
+        if(OperatingSystem.IsWindows())
+        {
+            Assert.Equal(Path.Join(projectDirectory, ".venv", "Scripts", "python.exe"), pythonProjectResource.Command);
+        }
+        else
+        {
+            Assert.Equal(Path.Join(projectDirectory, ".venv", "bin", "python"), pythonProjectResource.Command);
+        }
+
+        var commandArguments = await ArgumentEvaluator.GetArgumentListAsync(pythonProjectResource);
+
+        Assert.Equal(scriptName, commandArguments[0]);
+    }
+
+    [Fact]
+    public async Task AddPythonProjectWithInstrumentation_SwitchesExecutableToInstrumentationExecutable()
+    {
+        using var builder = TestDistributedApplicationBuilder.Create();
+
+        var (projectDirectory, pythonExecutable, scriptName) = CreateTempPythonProject(instrument: true);
+        
+        builder.AddPythonProject("pythonProject", projectDirectory, scriptName, virtualEnvironmentPath: ".venv");
+
+        var app = builder.Build();
+        var appModel = app.Services.GetRequiredService<DistributedApplicationModel>();
+        var executableResources = appModel.GetExecutableResources();
+
+        var pythonProjectResource = Assert.Single(executableResources);
+        var commandArguments = await ArgumentEvaluator.GetArgumentListAsync(pythonProjectResource);
+
+        if (OperatingSystem.IsWindows())
+        {
+            Assert.Equal(Path.Join(projectDirectory, ".venv", "Scripts", "opentelemetry-instrument.exe"), pythonProjectResource.Command);
+        }
+        else
+        {
+            Assert.Equal(Path.Join(projectDirectory, ".venv", "bin", "opentelemetry-instrument"), pythonProjectResource.Command);
+        }
+
+        Assert.Equal("--traces_exporter", commandArguments[0]);
+        Assert.Equal("otlp", commandArguments[1]);
+        Assert.Equal("--logs_exporter", commandArguments[2]);
+        Assert.Equal("console,otlp", commandArguments[3]);
+        Assert.Equal("--metrics_exporter", commandArguments[4]);
+        Assert.Equal("otlp", commandArguments[5]);
+        Assert.Equal(pythonExecutable, commandArguments[6]);
+        Assert.Equal(scriptName, commandArguments[7]);
```

Comment:
nit: This will also check the length, and avoid typos with the indices.

```suggestion
        Assert.Collection(commandArguments,
            arg => Assert.Equal("--traces_exporter", arg),
            arg => Assert.Equal("otlp", arg),
            arg => Assert.Equal("--logs_exporter", arg),
            arg => Assert.Equal("console,otlp", arg),
            arg => Assert.Equal("--metrics_exporter", arg),
            arg => Assert.Equal("otlp", arg),
            arg => Assert.Equal(pythonExecutable, arg),
            arg => Assert.Equal(scriptName, arg));
```